src/android/io/cozy/plugins/listlibraryitems/ListLibraryItems.java

```diff
@@ -325,13 +324,11 @@ protected Integer doInBackground(UploadFilePayload... uploadFilePayloads) {
                 OutputStream out = huc.getOutputStream();
                 buffer_size = Math.min(bytes_available, MAX_BUFFER_SZ);
                 buffer = new byte[buffer_size];
-                bytes_read = fis.read(buffer, 0, buffer_size);
                 total_bytes = bytes_read;
-                while (bytes_read > 0) {
+                while ((bytes_read = fis.read(buffer, 0, buffer_size)) != -1) {
                     out.write(buffer, 0, buffer_size);
                     bytes_available = fis.available();
                     buffer_size = Math.min(bytes_available, MAX_BUFFER_SZ);
-                    bytes_read = fis.read(buffer, 0, buffer_size);
                     total_bytes += bytes_read;
                     publishProgress(buffer_size, total_bytes, FILE_SZ);
```

Comment:
`buffer_size` as argument seems strange. Progress doesn't depend of buffer size.